vignettes/xcms.Rmd

```diff
@@ -1106,6 +1106,110 @@ properties of the mice analyzed (sex, age, litter mates etc).
 
 # Further data processing and analysis
 
+## Quality-based filtering of features
+
+When dealing with metabolomics results, it is often necessary to filter
+features based on certain criteria.
+These criteria are typically derived from statistical formulas applied to full
+rows of data, where each row represents a feature. The `filterFeatures`
+function provides a robust solution for filtering features based on these
+conventional quality assessment criteria. It supports multiple types of
+filtering, allowing users to tailor the filtering process to their specific
+needs, all controlled by the `filter` argument. This function and its
+implementations are applicable to both `XcmsExperiment` results objects and
+`SummarizedExperiment` objects.
+
+We will demonstrate how to use the `filterFeatures` function to perform quality
+assessment and filtering on both the faahko and res objects. The `filter`
+argument can accommodate various types of input, each determining the
+specific type of quality assessment and filtering to be performed.
+
+The `RsdFilter` enable users to filter features based on their coefficient of
+variation for a specified `threshold`. It is recommended to base the
+computation on quality control (QC) samples, as demonstrated below:
+
+```{r}
+# Set up parameters for RsdFilter
+rsd_filter <- RsdFilter(threshold = 0.3,
+                        qcIndex = sampleData(faahko)$sample_type == "QC")
+
+# Apply the filter to faakho object
+filtered_faahko <- filterFeatures(object = faahko, filter = rsd_filter)
+
+# Now apply the same strategy to the res object
+rsd_filter <- RsdFilter(threshold = 0.3, qcIndex = res$sample_type == "QC")
+filtered_res <- filterFeatures(object = res, filter = rsd_filter, assay = "raw")
+```
+
```

Comment:
Maybe say what happened by the call above.

*All features with an RSD (CV) larger than 0.3 in QC samples were thus removed from the data set.*